src/test/kotlin/org/wfanet/measurement/loadtest/dataprovider/EdpSimulatorTest.kt

```diff
@@ -297,50 +368,19 @@ class EdpSimulatorTest {
           EVENT_TEMPLATES,
           privacyBudgetManager
         )
+      val edpJob = launch { edpSimulator.process() }
+      delay(1000)
```

Comment:
Sorry to write here, something weird is going on, I can't review on reviewable. Lets do here for now.

This delay and cancelAndJoin pattern seems fragile. Did you see this somewhere in the codebase?